### PR TITLE
Add abort_with_status implementation

### PR DESCRIPTION
**Pull request by vpquant**
The `abort_with_status` method was added to the abstract class
`grpc.ServicerContext` in grpc 1.18.0, from which the `_OpenTracingServicerContext`
inherits.
(See https://github.com/grpc/grpc/pull/17481)
If no implementation for the new abstract method is provided, the `_OpenTracingServicerContext` class cannot be instantiated.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -12,9 +12,9 @@ jobs:
       # Download and cache dependencies
       - restore_cache:
           keys:
-          - v1-dependencies-{{ checksum "requirements-test.txt" }}
+          - v2-dependencies-{{ checksum "requirements-test.txt" }}
```
Comment (psiemens) [inline .circleci/config.yml]: Why the change from `v1` -> `v2`?

**Review by psiemens — COMMENTED** [review]
@rnburn @vpquant What's the status of this PR? I'm not a regular contributor, but I can confirm that this fix works with the latest version of gRPC. I'd love to see it merged!

File: .circleci/config.yml (hunk 1 above)
Comment (vpquant) [reply .circleci/config.yml]: The circleci pipeline was failing, because to the cache contained outdated packages. I modified the key prefix as recommended in the [circle-ci documentation](https://support.circleci.com/hc/en-us/articles/115015426888-Clear-project-cache).
However, there must be a better solution... (the same pipeline fails now). I don't know enough about CircleCI to fix this...

Comment (carlosalberto) [reply .circleci/config.yml]: I think we will switch to Travis as most OT projects use that one ;)

**Comment by carlosalberto** [comment]
Thanks a lot for this PR! I will now re-enable support for grpc >= 1.18.0 ;)

**Comment by psiemens** [comment]
Thanks for merging this @carlosalberto!
